File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int binarySearch(vector<int>& nums, int l, int r, int target) {
        while(l<=r)
        {
            int m = (l+r)/2;
            if(nums[m] == target)
                return m;
            else if(nums[m] > target)
                return binarySearch(nums,l,m-1,target);
            else
                return binarySearch(nums,m+1,r,target);
        }
        return -1;
    }
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;

        return binarySearch(nums,l,r,target);
        
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums: number[], k: number): number[] {
        const map = new Map<number, number>();

        for(const num of nums){
            map.set(num, (map.get(num) || 0) + 1)
        }

        const sorted = [...map.entries()].sort((a,b) => b[1] - a[1]);

        const r = new Array<number>();

        for(let i=0;i<k;i++){
            r.push(sorted[i][0])
        }
        return r;
    }
}
